From 5d785357ce3a74205d8330b6daf214980762f0a6 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 28 Dec 2021 14:39:42 +0800
Subject: [PATCH] Workaround gcc-11 errors

On ubuntu 20.04, install gcc-11
$ sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
$ sudo apt install -y gcc-11

1) make
$ S2N_LIBCRYPTO=openssl-1.1.1 BUILD_S2N=true TESTS=integration GCC_VERSION=11
$ source codebuild/bin/s2n_setup_env.sh
$ codebuild/bin/s2n_install_test_dependencies.sh
$ codebuild/bin/s2n_codebuild.sh

2) cmake
$ export CC="gcc-11"
$ cmake . -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Debug
$ cmake --build ./build -j $(nproc)

Upstream-Status: Submitted [https://github.com/aws/s2n-tls/pull/3160]
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 CMakeLists.txt | 3 +++
 s2n.mk         | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 127f9394..8c6904ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,6 +288,9 @@ set(CMAKE_C_FLAGS_DEBUGOPT "")
 
 target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
         -Wuninitialized -Wshadow -Wcast-qual -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
+        -Wno-error=array-parameter -Wno-error=discarded-qualifiers -Wno-error=analyzer-null-dereference
+        -Wno-error=analyzer-file-leak -Wno-error=analyzer-malloc-leak -Wno-error=analyzer-possible-null-dereference
+        -Wno-error=stringop-overflow -Wno-error=stringop-overread
         -Wno-missing-braces -Wa,--noexecstack
 )
 
diff --git a/s2n.mk b/s2n.mk
index ebd6bd9a..b90de84d 100644
--- a/s2n.mk
+++ b/s2n.mk
@@ -49,6 +49,9 @@ DEFAULT_CFLAGS += -pedantic -Wall -Werror -Wimplicit -Wunused -Wcomment -Wchar-s
                  -Wshadow  -Wcast-align -Wwrite-strings -fPIC -Wno-missing-braces\
                  -D_POSIX_C_SOURCE=200809L -O2 -I$(LIBCRYPTO_ROOT)/include/ \
                  -I$(S2N_ROOT)/api/ -I$(S2N_ROOT) -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security \
+                 -Wno-error=array-parameter -Wno-error=discarded-qualifiers -Wno-error=analyzer-null-dereference \
+                 -Wno-error=analyzer-file-leak -Wno-error=analyzer-malloc-leak -Wno-error=analyzer-possible-null-dereference \
+                 -Wno-error=stringop-overflow -Wno-error=stringop-overread \
                  -D_FORTIFY_SOURCE=2 -fgnu89-inline -fvisibility=hidden -DS2N_EXPORTS
 
 COVERAGE_CFLAGS = -fprofile-arcs -ftest-coverage
-- 
2.27.0

